### ISSUE TITLE
[Bug] Mangadex downloading the incorrect and random images

### ISSUE THREAD

**Issue by ahmadi3d** [HDoujinDownloader/HDoujinDownloader#216]
Try to download the first English chapter of the following link:
https://mangadex.org/title/5c9d1254-f074-4b5c-8795-6f0641b2597b/girl-and-science
You will see that it downloads the chapter but the images are random, some are actually the chapter but the rest are from different mangas!, Its a shuffle of different images!

**Comment by HDoujinDownloader** [comment]
I'm currently investigating this issue. It's not something I'm able to reproduce myself, but a lot of other users have reported the same problem. You can see my most recent comments on it [here](https://doujindownloader.com/forum/viewtopic.php?f=9&t=2400&p=5531#p5531) in the support forum.

I uploaded some fixes for the module as well as a program update to address some other potential causes. Try updating to the [latest pre-release](https://github.com/HDoujinDownloader/HDoujinDownloader/releases/download/v2023.6.13.0-rc.9/HDoujin.Downloader-2023.6.13.0-rc.9-win.zip) and let me know how it goes.

**Comment by LenXca** [comment]
> I'm currently investigating this issue. It's not something I'm able to reproduce myself, but a lot of other users have reported the same problem. You can see my most recent comments on it [here](https://doujindownloader.com/forum/viewtopic.php?f=9&t=2400&p=5531#p5531) in the support forum.
> 
> I uploaded some fixes for the module as well as a program update to address some other potential causes. Try updating to the [latest pre-release](https://github.com/HDoujinDownloader/HDoujinDownloader/releases/download/v2023.6.13.0-rc.9/HDoujin.Downloader-2023.6.13.0-rc.9-win.zip) and let me know how it goes.

Issue still remains :( I've checked a few other manga downloaders (hakuneko, tachiyomi etc) and they seem to be working fine.

**Comment by ahmadi3d** [comment]
@LenXca Yes, even after updating to the latest pre-release version, the problem still persists. However, I managed to solve it by using another downloader called "Mangadex Downloader," which is specifically designed for downloading from the Mangadex website. With this alternative downloader, I was able to successfully download the manga I wanted. Nevertheless, it would be more convenient if HDoujinDownloader could also support Mangadex downloads.

**Comment by LenXca** [comment]
@ahmadi3d Yea, that works too. Pretty much all other downloaders except HDoujin :(.
